hub/subscribe.go

```diff
@@ -87,16 +87,19 @@ func (h *Hub) initSubscription(w http.ResponseWriter, r *http.Request) (*Subscri
 	authorizedAlltargets, authorizedTargets := authorizedTargets(claims, false)
 	subscriber := NewSubscriber(authorizedAlltargets, authorizedTargets, topics, rawTopics, templateTopics, retrieveLastEventID(r))
 
-	if subscriber.LastEventID != "" {
-		h.sendMissedEvents(w, r, subscriber)
-	}
-
 	// Create a new channel, over which the hub can send can send updates to this subscriber.
 	updateChan := make(chan *serializedUpdate)
 
+	h.RLock()
 	// Add this client to the map of those that should
 	// receive updates
+	lastSentEventID := h.lastSentEventID
 	h.newSubscribers <- updateChan
+	h.RUnlock()
```

Comment:
there is still one concurency issue.
if a write is locked by this lock, the event will not be published to the subscriber because:
the event is sent after `lastSentEventID` and the subscriber is not yet registered